vllm/v1/sample/sampler.py

```diff
@@ -117,11 +117,17 @@ def sample(
         return sampled
 
     def compute_probs(self, logits: torch.Tensor,
-                      sampling_metadata: SamplingMetadata) -> torch.Tensor:
+                      sampling_metadata: SamplingMetadata,
+                      sample_lens: list[int]) -> torch.Tensor:
         if sampling_metadata.all_greedy:
             return logits
-        # Apply temperature. This is an in-place op changing logits.
-        logits = self.apply_temperature(logits, sampling_metadata.temperature)
+        assert sampling_metadata.temperature is not None
+        temperature = torch.repeat_interleave(
+            sampling_metadata.temperature,
+            torch.tensor(sample_lens,
+                         device=sampling_metadata.temperature.device))
+        temperature = temperature.unsqueeze(dim=1)
+        logits.div_(temperature)
```

Comment:
Here, we shouldn't update `logits` in place because `Sampler` will the logits (and apply the temperature again).
```suggestion
        logits = logits / temperature
```
Maybe also worth a comment.